cpp/src/arrow/array/array_test.cc

```diff
@@ -78,6 +78,24 @@ class TestArray : public ::testing::Test {
   MemoryPool* pool_;
 };
 
+TEST_F(TestArray, ValidateFullNullableList) {
+  auto f1 = field("f1", int32(), /*nullable=*/false);
+  auto ty = list(f1);
+  auto array = ArrayFromJSON(ty, "[[0, 1, 2, null], null, [4, 5]]");
+  ASSERT_RAISES(Invalid, array->ValidateFull());
+}
+
+TEST_F(TestArray, ValidateFullNullableFixedSizeList) {
+  auto f0 = field("f0", int32(), /*nullable=*/false);
+  auto type = fixed_size_list(f0, 2);
+  auto array_nonull = ArrayFromJSON(type, "[[0, 1], [3,4], [2, 3]]");
+  auto array = ArrayFromJSON(type, "[[0, 1], null, [2, 5]]");
```

Comment:
Hmm... the fact that this doesn't validate is more due to an implementation detail of `ArrayFromJSON`, I think.

Reply:
We can build the array manually here then?

```
auto inner = ArrayFromJSON(f0->type(), "[0, 1, 2, 3, 4, 5]");
std::shared_ptr<Buffer> null_bitmap;
ASSERT_OK(BitmapFromVector({1, 0, 1}, &null_bitmap));
auto array = std::make_shared<FixedSizeListArray>(type, /*length=*/3, inner);
ASSERT_OK(array->ValidateFull());
```